pkg/crypto/crypto_test.go

```diff
@@ -144,3 +200,38 @@ func generatePrivateKey() ([]byte, error) {
 	}
 	return buf.Bytes(), nil
 }
+
+// generateBuggyEncryptedString encrypt given string and returns it without modifying it
+// In older WMCO version encrypted string has extra tags. When we try to decrypt older
+// version( WMCO 4.xx and below) encrypted string in newer version we need to make sure we are
```

Comment:
This should read `WMCO 3.1.0 and below`

Reply:
we are targeting this fix in next release , right ? 4.x will be out as soon as we fix blocker issues. so I was kind of thinking this needs to go after 4.0 code is cut off.